openpype/tools/utils/widgets.py

```diff
@@ -606,3 +612,245 @@ def set_orientation(self, orientation):
+            header_bottom_row_widget
+        )
+        header_bottom_row_layout.setContentsMargins(0, 0, 0, 0)
+        header_bottom_row_layout.setSpacing(2)
+        header_bottom_row_layout.addWidget(filter_input, 2)
+        header_bottom_row_layout.addWidget(clear_filter_btn, 1)
+
+        header_layout = QtWidgets.QVBoxLayout(self)
+        header_layout.setContentsMargins(0, 0, 0, 0)
+        header_layout.addWidget(header_top_row_widget)
+        header_layout.addWidget(header_bottom_row_widget)
+
+        # Make header widgets expand vertically if there is a place
+        for widget in (
+            title_label,
+            current_item_btn,
+            refresh_btn,
+            clear_filter_btn,
+            filter_input,
+        ):
+            size_policy = widget.sizePolicy()
+            size_policy.setVerticalPolicy(size_policy.MinimumExpanding)
+            widget.setSizePolicy(size_policy)
+
+        self._show_search_bar = show_search_bar
+        self._show_refresh_button = show_refresh_button
+
+        self._filter_input = filter_input
+        self._refresh_btn = refresh_btn
+        self._current_item_btn = current_item_btn
+
+        clear_filter_btn.clicked.connect(self._on_clear_filter_btn_released)
+        filter_input.currentTextChanged.connect(
+            parent_item_widget._on_filter_text_change
+        )
+        refresh_btn.clicked.connect(parent_item_widget.refresh)
+        current_item_btn.clicked.connect(
+            parent_item_widget._on_current_item_click
+        )
+
+        self._header_bottom_row_widget = header_bottom_row_widget
+        self._refresh_btn = refresh_btn
+        self._header_layout = header_layout
+
+    @QtCore.Slot()
+    def _on_clear_filter_btn_released(self):
+        self._filter_input.setCurrentText("")
+
+    def showEvent(self, event):
+        """
+        Use showEvent to add top and bottom margins to have labels
+        match even if search and refresh button are hidden.
+        This will work out the expected height ratio of the
+        header as there is slight variance depending on
+        environment where this code is run.
+        """
+
+        super().showEvent(event)
+        layout = self.layout()
+        layout.update()
+        layout.activate()
+        layout.invalidate()
+        show_search_bar = self._show_search_bar
+        show_refresh_button = self._show_refresh_button
+        if not show_search_bar:
+            self._header_bottom_row_widget.hide()
+
+        if not show_refresh_button:
+            self._refresh_btn.hide()
+
+        if not (show_search_bar and show_refresh_button):
+            height = self.height()
+            # these ratios were calculated using standard resolution
+            # and scaling:
+            top = int(height * 0.133333)
+            bottom = int(height * 0.533333)
+            self._header_layout.setContentsMargins(0, top, 0, bottom)
+
+
+class ItemViewWidget(QtWidgets.QWidget):
```

Comment:
I quite don't like this approach. If we'll need to change how assets or tasks widgets works we'll be blocked by the fact they have this parent. I would keep assets widget and task widget as 2 separated classes (from top to bottom). If we'll need change one of them in some way it's more complicated. Also like I've mentioned I don't think tasks widget should have headers -> please create a ticket to propose that.

Reply:
Fair request to split this up further - I just won't have the time to do it for a while as I need to get back to my in house work.
I will open a ticket about the changes we have made to the launcher and workfiles ui when I get a spare moment too.

I appreciate the time taken to review and give feedback.